osmhelper/osm_connector.py

```diff
@@ -0,0 +1,325 @@
+            caching if possible.
+
+        :return routes: A dictionary of RouteMaster objects with related
+            RouteVariant objects constituting the tree of data.
+
+        """
+        # Preferably return cached data about routes
+        if refresh is False:
+            # Check if routes data is already built in this object
+            if not self.routes:
+                # If not, try to get routes data from file cache
+                self.routes = Cache.read_data('routes')
+            # Return cached data if found
+            if bool(self.routes):
+                return self.routes
+
+        # No cached data was found or refresh was forced
+        print "Query and build fresh data for routes"
+
+        # Obtain raw data about routes from OpenStreetMap
+        result = self._query_routes()
+
+        # Pre-sort relations by type
+        route_masters = {}
+        route_variants = {}
+        for relation in result.relations:
+            if relation.tags["type"] == "route_master":
+                route_masters[relation.id] = relation
+            else:
+                route_variants[relation.id] = relation
+
+        # Build routes from master relations
+        for rmid, route_master in route_masters.iteritems():
+            members = OrderedDict()
+
+            # Build route variant members
+            for member in route_master.members:
+
+                if member.ref in route_variants:
+                    rv = route_variants.pop(member.ref)
+                    members[rv.tags['ref']
+                            ] = self._build_route_variant(rv, result)
+
+                # Route master member was already used before or is not valid
+                else:
+                    rv = result.get_relations(member.ref)
+                    if bool(rv):
+                        print "Route variant was assigned again:"
+                        print "http://osm.org/relation/" + str(rv.id)
+                        members[rv.tags['ref']
+                                ] = self._build_route_variant(rv, result)
+                    else:
+                        print "Member relation is not a valid route variant:"
+                        print "http://osm.org/relation/" + str(member.ref)
+
+            rm = self._build_route_master(route_master, members)
+            # Make sure ref number is not already taken
+            if rm.ref in self.routes:
+                sys.stderr.write("Ref:" + rm.ref + "already in routes\n")
+            else:
+                self.routes[rm.ref] = rm
+
+        # Build routes from variants (missing master relation)
+        for rvid, route_variant in route_variants.iteritems():
+            sys.stderr.write("Routevariant without masters\n")
+            rv = self._build_route_variant(route_variant, result)
+            # Make sure ref number is not already taken
+            if rv.ref in self.routes:
+                sys.stderr.write("Ref:" + rv.ref + "already in routes\n")
```

Comment:
Same to the comment above. All this proper error handling would be soo much easier within #30. So, let's not forget it, but also not make it part of the scope of this PR. Before the problem existed the same way, but not even any message was advised.